client/state/index.js

```diff
@@ -111,26 +115,113 @@ export const reducer = combineReducers( {
+		this.reduxStore = null;
+	}
+
+	createReduxStore( initialState = {} ) {
+		const isBrowser = typeof window === 'object';
+		const isAudioSupported = typeof window === 'object' && typeof window.Audio === 'function';
+
+		const middlewares = [
+			thunkMiddleware,
+			noticesMiddleware,
+			isBrowser && require( './analytics/middleware.js' ).analyticsMiddleware,
+			isBrowser && require( './data-layer/wpcom-api-middleware.js' ).default,
+			isAudioSupported && require( './audio/middleware.js' ).default,
+		].filter( Boolean );
+
+		const enhancers = [
+			isBrowser && window.app && window.app.isDebug && consoleDispatcher,
+			applyMiddleware( ...middlewares ),
+			isBrowser && sitesSync,
+			isBrowser && window.devToolsExtension && window.devToolsExtension()
+		].filter( Boolean );
+
+		const reducer = this.getCombinedReducer();
+		this.reduxStore = compose( ...enhancers )( createStore )( reducer, initialState );
+		return this.reduxStore;
+	}
+
+	addReducer( name, reducerFunc ) {
+		if ( this.reducers[ name ] !== undefined ) {
+			throw new Error( `addReducer(): name ${ name } already in use` );
+		}
+
+		this.setReducers( { ...this.reducers, [ name ]: reducerFunc } );
+	}
+
+	removeReducer( name ) {
+		const { [ name ]: removedReducer, ...remainingReducers } = this.reducers;
+
+		if ( ! removedReducer ) {
+			debug( 'removeReducer(): name not found' );
+			return;
+		}
+
+		this.setReducers( remainingReducers );
+	}
+
+	getCombinedReducer() {
+		const reducers = this.reducers;
+		return combineReducers( reducers );
+	}
+
+	setReducers( reducers ) {
+		this.reducers = reducers;
+
+		if ( this.reduxStore ) {
+			this.reduxStore.replaceReducer( this.getCombinedReducer() );
+		}
+	}
 }
+
+const calypsoStore = new CalypsoStore();
+
+/**
+ * Named export function for creating a redux store.
+ * This creates a new redux store within the calypsoStore instance.
+ *
+ * @param { object } initialState Redux state with which to initialize (optional).
+ * @return { object } The redux store that was created.
+ */
+export function createReduxStore( initialState ) {
+	return calypsoStore.createReduxStore( initialState );
+}
+
+/**
+ * Named export for adding a reducer dynamically to the store.
+ *
+ * @param { string } name The top-level context name for the reducer.
+ * @param { function } func The reducer function, in the format of ( state, action ), returning state.
+ */
+export function addReducer( name, func ) {
+	calypsoStore.addReducer( name, func );
+}
+
+/**
+ * Named export for adding a reducer dynamically to the store.
+ *
+ * @param { string } name The top-level context name for the reducer.
+ */
+export function removeReducer( name ) {
+	calypsoStore.removeReducer( name );
+}
+
+/**
+ * Named export for getting the combined reducer from the store.
+ *
+ * @return { function } The currently configured reducer chain.
+ */
+export function getCombinedReducer() {
+	return calypsoStore.getCombinedReducer();
+}
+
```

Comment:
To me, the explicitness is worth the perceived duplication. Those function are the main proposed interface for developers and being able to find them defined in the source can be very useful when navigating the codebase.